api/v1beta1/common_openstackbaremetalset.go

```diff
@@ -123,17 +123,35 @@ func VerifyBaremetalSetScaleUp(log logr.Logger, instance *OpenStackBaremetalSet,
 	if newBmhsNeededCount > 0 {
 		// We have new replicas requested, so search for baremetalhosts that don't have consumerRef or Online set
 
+		labelStr := ""
+
+		if len(instance.Spec.BmhLabelSelector) > 0 {
+			labelStr = fmt.Sprintf(" with labels %v", instance.Spec.BmhLabelSelector)
+			labelStr = strings.Replace(labelStr, "map[", "[", 1)
+		}
+
+		log.Info(fmt.Sprintf("Attempting to find %d BaremetalHost(s)%s for scale-up of OpenStackBaremetalSet %s", newBmhsNeededCount, labelStr, instance.Name))
+
 		for _, baremetalHost := range allBmhs.Items {
-			if baremetalHost.Spec.Online || baremetalHost.Spec.ConsumerRef != nil {
-				continue
+			mismatch := false
+
+			if baremetalHost.Spec.Online {
+				log.Info(fmt.Sprintf("BaremetalHost %s cannot be used because it is already online", baremetalHost.ObjectMeta.Name))
+				mismatch = true
```

Comment:
does it make sense to continue here? I guess if a BMH is online it has a consumerRef and we do not have to check for hardware req, or can there be a situation where this is not the case?

Reply:
The reason I am not just continuing is that the customer asked specifically for a detailed list of which things were preventing the BMH from being selected.  So my aim was to gather everything that applied as opposed to just logging the first one.